injectproxy/rules.go

```diff
@@ -207,11 +207,8 @@ func (r *routes) filterRules(lvalues []string, resp *apiResponse) (interface{},
 	for _, rg := range rgs.RuleGroups {
 		var rules []rule
 		for _, rule := range rg.Rules {
-			for _, lbl := range rule.Labels() {
-				if lbl.Name == r.label && slices.Contains(lvalues, lbl.Value) {
-					rules = append(rules, rule)
-					break
-				}
+			if rule.Labels().Has(r.label) && slices.Contains(lvalues, rule.Labels().Get(r.label)) {
```

Comment:
This would be more efficient calling just `Get` and checking for an non-empty result.

Reply:
That makes sense! Thank you

Reply:
changed it!